drivers/ad5683/ad5683.c

```diff
@@ -0,0 +1,334 @@
+ * @return ret - The result of the write procedure.
+*/
+int32_t ad5683_update_register(struct ad5683_dev *dev)
+{
+	return ad5683_set_shift_reg(dev, AD5683_CMD_UPDATE_DAC, 0);
+}
+
+/**
+ * @brief Write DAC and Input Register
+ *
+ * @param dev     - The device structure.
+ * @param data    - Desired value to be written in register.
+ *
+ * @return ret - The result of the write procedure.
+*/
+int32_t ad5683_write_dac_input_register(struct ad5683_dev *dev, uint16_t data)
+{
+	uint8_t data_offset = MAX_RESOLUTION - chip_info[dev->act_device].resolution;
+
+	return ad5683_set_shift_reg(dev, AD5683_CMD_WR_DAC_REG, data << data_offset);
+}
+
+/**
+ * @brief Read back Input Register
+ *
+ * @param dev     - The device structure.
+ *
+ * @return read_back_data - value read from register.
+*/
+uint32_t ad5683_read_back_register(struct ad5683_dev *dev)
+{
+
+	uint32_t read_back_data = 0;
+	uint32_t offset = MAX_RESOLUTION - chip_info[dev->act_device].resolution;
+
+	if(chip_info[dev->act_device].communication == SPI) {
+		ad5683_set_shift_reg(dev, AD5683_CMD_RB_IN_REG, 0);
+		read_back_data = ad5683_set_shift_reg(dev, AD5683_CMD_NOP, 0);
+		read_back_data = (RB_DATA_MASK & read_back_data) >> (offset + LSB_OFFSET);
+	} else
+		return FAILURE;
+
+	return read_back_data;
+}
+
+/**
+ * @brief Set Power-down mode.
+ *
+ * @param dev     - The device structure.
+ * @param mode    - Power-down operation modes.
+ *                  Accepted values:
+ *                  'AD5683_PWRM_NORMAL' - Normal Mode
+ *                  'AD5683_PWRM_1K' - Power-down mode 1kOhm to GND
+ *                  'AD5683_PWRM_100K' - Power-down mode 100kOhm to GND
+ *                  'AD5683_PWRM_THREESTATE' - Three-State
+ *
+ * @return ret - The result of the write procedure.
+*/
+int32_t ad5683_power_mode(struct ad5683_dev *dev, uint8_t mode)
+{
+	return ad5683_set_shift_reg(dev, AD5683_CMD_WR_CTRL_REG,
+				    AD5683_CTRL_PWRM(mode));
+}
+
+/**
+ * @brief Software reset
+ *
+ * @param dev - The device structure.
+ *
+ * @return ret - The result of the write procedure.
+*/
+int32_t ad5683_software_reset(struct ad5683_dev *dev)
+{
+	return ad5683_set_shift_reg(dev, AD5683_CMD_WR_CTRL_REG,
+				    AD5683_SW_RESET(AD5683_SW_RESET_EN));
+}
+
+/**
+ * @brief Set Internal reference
+ *
+ * @param dev   - The device structure.
+ * @param value - The internal reference register value
+ *                Example : 'AD5683_INTREF_EN' - enable internal reference
+ *                          'AD5683_INTREF_DIS' - disable internal reference
+ *
+ * @return ret - The result of the write procedure.
+*/
+int32_t ad5683_internal_reference(struct ad5683_dev *dev, uint8_t value)
+{
+	return ad5683_set_shift_reg(dev, AD5683_CMD_WR_CTRL_REG,
+				    AD5683_CTRL_INT_REF(value));
+}
+
+/**
+ * @brief Set DCEN (daisy-chain enable)
+ *
+ * @param dev   - The device structure.
+ * @param value - Enable or disable daisy-chain mode
+ *                Example : 'AD5683_DC_EN' - daisy-chain enable
```

Comment:
related to commit https://github.com/analogdevicesinc/no-OS/pull/104/commits/d1914dad852f657681586dd757949bae0c63a60e#r221907052  you could just do  `uint8_t enable`, where 0 disables and non-zero enables